fastlane/lib/fastlane/plugin_manager.rb

```diff
@@ -2,35 +2,43 @@ module Fastlane
   class PluginManager
     require 'bundler'
 
-    PLUGINSFILE_NAME = "Plugins"
+    PLUGINSFILE_NAME = "Plugins".freeze
+    DEFAULT_GEMFILE_PATH = "Gemfile".freeze
+    GEMFILE_SOURCE_LINE = "source \"https://rubygems.org\"\n"
 
     def gemfile_path
-      # This is pretty important, since we don't know what kind of
-      # Gemfile the user has (e.g. Gemfile, gems.rb, or custom env variable)
-      Bundler::SharedHelpers.default_gemfile.to_s
+      begin
+        # This is pretty important, since we don't know what kind of
+        # Gemfile the user has (e.g. Gemfile, gems.rb, or custom env variable)
+        Bundler::SharedHelpers.default_gemfile.to_s
+      rescue Bundler::GemfileNotFound
+        nil
+      end
     end
 
     def pluginsfile_path
       File.join(FastlaneFolder.path, PLUGINSFILE_NAME)
     end
 
     def gemfile_content
-      return File.read(gemfile_path) if File.exist?(gemfile_path)
-      nil
+      File.read(gemfile_path) if gemfile_path && File.exist?(gemfile_path)
     end
 
     def pluginsfile_content
-      return File.read(pluginsfile_path) if File.exist?(pluginsfile_path)
-      nil
+      File.read(pluginsfile_path) if File.exist?(pluginsfile_path)
     end
 
     def add_dependency(plugin_name)
       unless (pluginsfile_content || "").include?(plugin_name)
         content = pluginsfile_content || "# Autogenerated by fastlane\n\n"
-        content += "gem '#{plugin_name}'\n"
+        # TODO: obviously need to stop hardcoding a git URL!
+        content += "gem '#{plugin_name}', git: 'git@github.com:fastlane/#{plugin_name}.git'\n"
```

Comment:
By default, we'll just have `gem "fastlane_name"`, right? The git url should only be used when the owner doesn't push their gem to RubyGems